Source/Charts/Charts/BarLineChartViewBase.swift

```diff
@@ -753,7 +753,7 @@ open class BarLineChartViewBase: ChartViewBase, BarLineScatterCandleBubbleChartD
                 
                 if ((h === nil && lastHighlighted !== nil) ||
                     (h !== nil && lastHighlighted === nil) ||
-                    (h !== nil && lastHighlighted !== nil && !h!.isEqual(lastHighlighted)))
+                    (h !== nil && lastHighlighted !== nil && !(h?.isEqual(lastHighlighted) ?? false)))
```

Comment:
Here we've already checked that `h != nil` in the AND statement. If `h == nil` escape early from the AND statement, so it is safe to force unwrap `h`

Reply:
Currently there is an issue in the last statement. If the whole expression was evaluated, then in the last component (`!(h?.isEqual(lastHighlighted) ?? false)`) you would return true when `h == nil` which is not what we want.

Reply:
done